src/main/java/com/streamr/client/StreamrRESTClient.java

```diff
@@ -123,4 +126,9 @@ public UserInfo getUserInfo() throws IOException {
         HttpUrl url = HttpUrl.parse(options.getRestApiUrl() + "/users/me");
         return get(url, userInfoJsonAdapter);
     }
+
+    public List<String> getPublishers(String streamId) throws IOException {
```

Comment:
Even if only the `stream.getId()` is used? This method is called by
```
cache.getPublishers(msg.getStreamId())
```
So it should be replaced by
```
Stream stream = cache.getStream(msg.getStreamId())
cache.getPublishers(stream)
```
so unless it is cached, it would make a request to `/streams/${id}` even though it is useless.